src/util/data.js

```diff
@@ -1,70 +1,154 @@
-import env from 'env-var';
+import FHIR from "fhirclient";
 
-const types = {
-    error: "errorClass",
-    info: "infoClass",
-    debug: "debugClass",
-    warning: "warningClass"
+const clearQuestionnaireResponses = ({ ehrUrl, defaultUser, access_token }, consoleLog) => (_event) => {
+    console.log("Clear QuestionnaireResponses from the EHR: " + ehrUrl + " for author " + defaultUser);
+    const client = FHIR.client({ 
+        serverUrl: ehrUrl, 
+        ...(access_token ? { tokenResponse: access_token } : {})
+    });
+    client
+        .request("QuestionnaireResponse?author=" + defaultUser, { flat: true })
+        .then((result) => {
+            console.log(result);
+            result.forEach((resource) => {
+                console.log(resource.id);
+                client
+                    .delete("QuestionnaireResponse/" + resource.id)
+                    .then((result) => {
+                        consoleLog("Successfully deleted QuestionnaireResponse " + resource.id + " from EHR", types.info);
+                        console.log(result);
+                    })
+                    .catch((e) => {
+                        console.log("Failed to delete QuestionnaireResponse " + resource.id);
+                        console.log(e);
+                    });
+            });
+        })
+        .catch((e) => {
+            console.log("Failed to retrieve list of QuestionnaireResponses");
+            console.log(e);
+        });
+};
+
+const resetPims = ({ pimsUrl }, consoleLog) => (_event) => {
+    let url = new URL(pimsUrl);
+    const resetUrl = url.origin + "/doctorOrders/api/deleteAll";
+    console.log("reset pims: " + resetUrl);
+    
+    fetch(resetUrl, {
+        method: 'DELETE',
+    })
+    .then(response => {
+        console.log("Reset pims: ");
+        console.log(response);
+        consoleLog("Successfully reset pims database", types.info);
+    })
+    .catch(error => {
+        console.log("Reset pims error: ");
+        consoleLog("Server returned error when resetting pims: ", types.error);
+        consoleLog(error.message);
+        console.log(error);
+    });
 }
 
-const headers = {
-    "ehrUrl": {
-        "display": "EHR Server",
-        "value": (env.get('REACT_APP_EHR_SERVER').asString()),
-        "key": "ehrUrl"
+const resetRemsAdmin = ({ cdsUrl } , consoleLog) => (_event) => {
+    let url = new URL(cdsUrl);
+    const resetUrl = url.origin + "/etasu/reset";
+
+    fetch(resetUrl, {
+        method: 'POST',
+    })
+    .then(response => {
+        console.log("Reset rems admin etasu: ");
+        console.log(response);
+        consoleLog("Successfully reset rems admin etasu", types.info);
+    })
+    .catch(error => {
+        console.log("Reset rems admin error: ");
+        consoleLog("Server returned error when resetting rems admin etasu: ", types.error);
+        consoleLog(error.message);
+        console.log(error);
+    });
+}
+
+
+const headerDefinitions = {
```

Comment:
these definitions are a little disjointed.  They are passed to SettingsBox, which use them to associate a key with a state value in RequestBox, but the state value in RequestBox is not based on these definitions.  

We'd probably be better off getting rid of this definition and making a `settings` object in the RequestBox state that we update.

Reply:
I think I am lost on your comment, in terms of specific code changes. RequestBox uses the state from the parent component RequestBuilder, which is then passed into and mutated by SettingsBox (so RequestBox and SettingsBox are siblings).

Reply:
I do notice that despite changing the "default user", I'm still seeing pra1234 in the prefetch (from RequestBox, which uses the `submit_info` function from RequestBuilder which does have a `this.state.defaultUser`.

Reply:
![image](https://github.com/mcode/request-generator/assets/33106214/52c7cc6e-a530-411b-a0e1-10119f19887a)

Reply:
Sorry, I said RequestBox when I meant RequestBuilder.  The basic idea here is that the name of the key is being defined in two different places.  This should be unnecessary.  SettingsBox could potentially make an input that updates a state that doesn't actually exist and it would produce no error.  

In terms of code changes... this would basically require an overhaul of the whole system, so that we have only one "source of truth" which defines the RequestBox's state AND the inputs that get automatically generated.  It's probably not worth it.

Reply:
Maybe we can create a ticket/sub-ticket (?) as part of refactoring/making tests for request generator (would that be an epic? I am unsure of the right terminology).

Reply:
make a low priority ticket for refactoring the settings, call it a day

Reply:
Done, the ticket is https://jira.mitre.org/browse/REMS-540